src/main/python/deep_qa/layers/entailment_models/tuple_match.py

```diff
@@ -0,0 +1,180 @@
+
+        Output:
+            - entailment score, shape ``(batch, 1)``
+
+        Parameters
+        ----------
+        params: dict
+            Specifies each of the following (with defaults if unspecified):
+            - num_entailment_features : int, default=4
+                The number of features that are generated by the entailment and passed into the shallow NN.
+            - num_hidden_layers : int, default=1
+                Number of hidden layers in the shallow NN.
+            - hidden_layer_width : int, default=4
+                The number of nodes in each of the NN hidden layers.
+            - hidden_layer_activation : string, default='relu'
+                The activation of the NN hidden layers
+            - use_biases: int, default=1
+                A 0/1 corresponding to disabling/enabling biases in each of the hidden layers
+        """
+
+    def __init__(self, params: Dict[str, Any]):
+        """
+        Construct a new TupleMatch layer.
+
+        Parameters
+        ----------
+
+        """
+        self.input_dim = None
+        self.supports_masking = True
+        # Parameters for the shallow neural network
+        self.num_entailment_features = params.pop('num_entailment_features', 4)
+        self.num_hidden_layers = params.pop('num_hidden_layers', 1)
+        self.hidden_layer_width = params.pop('hidden_layer_width', 4)
+        self.hidden_layer_activation = params.pop('hidden_layer_activation', 'relu')
+        self.use_biases = bool(params.pop('use_bias', 1))
+        # NN layers
+        self.hidden_layers = None
+        self.score_layer = None
+        super(TupleMatch, self).__init__(**params)
+
+    def score_tuple_entailment(self, combined_tuple_input):
+        """
+        Here we take the combined tuple-based entailment input and decide whether it is true or false. The
+        combined input has shape (batch_size, combined_input_dim).  We do not know what
+        combined_input_dim is, as it depends on the combiner.
+        """
+        hidden_input = combined_tuple_input
+        for layer in self.hidden_layers:
+            hidden_input = layer(hidden_input)
+        score_output = self.score_layer(hidden_input)
+        return score_output
+
+
+    def get_output_shape_for(self, input_shapes):
+        #pylint: disable=unused-argument
+        return (1,)
+
+
+    def build(self, input_shape):
+        # Build the dense layer(s)
+        self.hidden_layers = []
+        dense_input_shape = (None, self.num_entailment_features)
+        for i in range(self.num_hidden_layers):
+            curr_dense = Dense(output_dim=self.hidden_layer_width,
+                               activation=self.hidden_layer_activation,
+                               bias=self.use_biases,
+                               name='tuple_match_hidden_layer_%d' % i)
+            curr_dense.build(input_shape=dense_input_shape)
```

Comment:
re: second comment again -- sorry, I really don't understand what's happening in the code you pointed me to... if you want me to use it, can you direct me a little? (sorry!)